common/unsafe/src/main/java/org/apache/spark/unsafe/memory/OffHeapMemoryBlock.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe.memory;
+
+import org.apache.spark.unsafe.Platform;
+
+public class OffHeapMemoryBlock extends MemoryBlock {
+  private Object directBuffer;
+  private final long address;
+
+  static public final OffHeapMemoryBlock NULL = new OffHeapMemoryBlock(null, 0, 0);
+
+  public OffHeapMemoryBlock(Object directBuffer, long address, long size) {
+    super(null, address, size);
+    this.address = address;
+    this.directBuffer = directBuffer;
```

Comment:
shall we just `assert directBuffer == null` and remove the `directBuffer` member?